### ISSUE TITLE
Test CLEARCHAT

### ISSUE THREAD

**Issue by thePlebDev** [thePlebDev/Clicker#307]
# Proposed change
- I want to put the CLEARCHAT parsing code into a utility object and test it

# Why is this important
- This will make the code more testable and readable.
- Also, this will allow us to verify the text that we are parsing.
- First off, we will move everything from the `text.contains` conditional into a new object we will call the parsing engine.

# Additional context
- we need to do this for all the objects that we are parsing
- eventually I want the parsing engine to handle everything, which includes the `text.contains` conditionals

**Comment by thePlebDev** [comment]
- I want to be able to test these:
```
Examples
The following message indicates that all chat messages were removed from the dallas chat room.

:tmi.twitch.tv CLEARCHAT #dallas

The following message indicates that all chat messages that ronni posted in the dallas chat room were removed.

:tmi.twitch.tv CLEARCHAT #dallas :ronni
```

**Comment by thePlebDev** [comment]
- closed with #313
